src/modules/powerrename/lib/Settings.cpp

```diff
@@ -470,12 +359,205 @@ void CRenameMRU::_FreeMRUList()
+    settings.flags = flags;
+}
+
+const std::wstring& CSettings::GetSearchText() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.searchText;
+}
+
+void CSettings::SetSearchText(const std::wstring& text)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.searchText = text;
+}
+
+const std::wstring& CSettings::GetReplaceText() const
+{
+    std::scoped_lock lock{ dataLock };
+    return settings.replaceText;
+}
+
+void CSettings::SetReplaceText(const std::wstring& text)
+{
+    std::scoped_lock lock{ dataLock };
+    settings.replaceText = text;
+}
+
+void CSettings::LoadPowerRenameData()
+{
+    std::scoped_lock lock{ dataLock };
+
+    if (!std::filesystem::exists(jsonFilePath))
+    {
+        MigrateSettingsFromRegistry();
+    }
+    else
+    {
+        ParseJsonSettings(GetPersistPowerRenameJSON());
+
+        SavePowerRenameData();
+    }
+}
+
+void CSettings::SavePowerRenameData() const
+{
+    std::scoped_lock lock{ dataLock };
+
+    json::JsonObject jsonData;
+
+    jsonData.SetNamedValue(c_enabled,                 json::value(settings.enabled));
+    jsonData.SetNamedValue(c_showIconOnMenu,          json::value(settings.showIconOnMenu));
+    jsonData.SetNamedValue(c_extendedContextMenuOnly, json::value(settings.extendedContextMenuOnly));
+    jsonData.SetNamedValue(c_persistState,            json::value(settings.persistState));
+    jsonData.SetNamedValue(c_mruEnabled,              json::value(settings.MRUEnabled));
+    jsonData.SetNamedValue(c_maxMRUSize,              json::value(settings.maxMRUSize));
+    jsonData.SetNamedValue(c_flags,                   json::value(settings.flags));
+    jsonData.SetNamedValue(c_searchText,              json::value(settings.searchText));
+    jsonData.SetNamedValue(c_replaceText,             json::value(settings.replaceText));
+
+    json::to_file(jsonFilePath, jsonData);
+}
+
+json::JsonObject CSettings::GetPersistPowerRenameJSON()
+{
+    std::scoped_lock lock{ dataLock };
+
+    auto jsonData = json::from_file(jsonFilePath);
+    return jsonData ? jsonData.value() : json::JsonObject();
+}
+
+void CSettings::MigrateSettingsFromRegistry()
+{
+    std::scoped_lock lock{ dataLock };
+
+    settings.enabled                 = GetRegBoolean(c_enabled, true);
+    settings.showIconOnMenu          = GetRegBoolean(c_showIconOnMenu, true);
+    settings.extendedContextMenuOnly = GetRegBoolean(c_extendedContextMenuOnly, false); // Disabled by default.
+    settings.persistState            = GetRegBoolean(c_persistState, true);
+    settings.MRUEnabled              = GetRegBoolean(c_mruEnabled, true);
+    settings.maxMRUSize              = GetRegNumber(c_maxMRUSize, 10);
+    settings.flags                   = GetRegNumber(c_flags, 0);
+    settings.searchText              = GetRegString(c_searchText);
+    settings.replaceText             = GetRegString(c_replaceText);
+}
+
+void CSettings::ParseJsonSettings(const json::JsonObject& jsonSettings)

```

Comment:
please remove this method, since you already initialize `Settings` class with the same values.